core/src/main/java/org/axonframework/eventsourcing/EventSourcingRepository.java

```diff
@@ -467,4 +181,184 @@ protected void doDeleteWithLock(EventSourcedAggregate<T> aggregate) {
+         * locking strategy is not compatible with a caching approach.
+         */
+        @Override
+        public Builder<T> lockFactory(LockFactory lockFactory) {
+            super.lockFactory(lockFactory);
+            return this;
+        }
+
+        /**
+         * Sets the {@link EventStore} that holds the event stream this repository needs to event source an Aggregate.
+         *
+         * @param eventStore an {@link EventStore} that holds the event stream this repository needs to event source
+         *                   an Aggregate
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder<T> eventStore(EventStore eventStore) {
+            assertNonNull(eventStore, "EventStore may not be null");
+            this.eventStore = eventStore;
+            return this;
+        }
+
+        /**
+         * Sets the {@link SnapshotTriggerDefinition} specifying when to trigger a snapshot for an Aggregate contained
+         * in this repository.
+         *
+         * @param snapshotTriggerDefinition a {@link SnapshotTriggerDefinition} specifying when to trigger a snapshot
+         *                                  for an Aggregate contained in this repository
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder<T> snapshotTriggerDefinition(SnapshotTriggerDefinition snapshotTriggerDefinition) {
+            assertNonNull(snapshotTriggerDefinition, "SnapshotTriggerDefinition may not be null");
+            this.snapshotTriggerDefinition = snapshotTriggerDefinition;
+            return this;
+        }
+
+        /**
+         * Sets the {@link AggregateFactory} used to create new Aggregate instances. Either this field or the
+         * {@link #aggregateType(Class)} should be provided to correctly instantiate an AggregateFactory for this
+         * {@link EventSourcingRepository}.
+         *
+         * @param aggregateFactory the {@link AggregateFactory} used to create new Aggregate instances
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder<T> aggregateFactory(AggregateFactory<T> aggregateFactory) {
+            assertNonNull(aggregateFactory, "AggregateFactory may not be null");
+            this.aggregateFactory = aggregateFactory;
+            return this;
+        }
+
+        /**
+         * Sets the {@link RepositoryProvider} which services repositories for specific aggregate types.
+         *
+         * @param repositoryProvider a {@link RepositoryProvider} servicing repositories for specific aggregate types
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder<T> repositoryProvider(RepositoryProvider repositoryProvider) {
+            this.repositoryProvider = repositoryProvider;
```

Comment:
Assertion?

Reply:
It used to be nullable when you introduced it, so I left it that way.